---
title: "MP_SRF_amplicons_fraction_18_and_16S_diversity_analysis"
author: "lrubinat"
date: "05/05/2016"
output:
  html_document:
    theme: united
    toc: yes
  pdf_document:
    highlight: zenburn
    toc: yes
---

<!--- INITIALIZATION
```{r, echo=FALSE}
#error hook to kill knitr in case of errors
library(knitr)
knit_hooks$set(error = function(x, options) stop(x))
opts_chunk$set(cache=TRUE, autodep=TRUE)
```
--->

# 1) 18S amplicons

## 1.1) Data overview

Let's read the dataset and remove the samples containing less than 11770 reads:

``` {r load_data, echo=FALSE, message=FALSE}
setwd("/home/laura/Documents/TFM/genwork/data_analysis/MP_SRF_amplicons_fraction_18S_and_16S_diversity_analysis/")

#read data 
tb18_tax <- read.table(file="/home/laura/Documents/TFM/home/data/MALASPINA/Malaspina_18S_Surface/amplicons_fraction/MP_18S_SRF_MAS_BM_SILVA_classif_tab_filtered_amplicons_fraction_sorted.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(tb18_tax) # 22702    18
tb18_tax[1:5,1:5]

#row names = OTU name (option A)
row.names(tb18_tax)<-tb18_tax[,1]

#row names = row number (option B)
#rownames(otu_tb18) <- 1:nrow(otu_tb18)

tb18_tax<-tb18_tax[,-1]
tb18_tax[is.na(tb18_tax)]<-"NA"

dim(tb18_tax)
tb18_tax[1:5,1:5]

#table with taxonomi classification alone
tb18_class <- tb18_tax[,11:17]
dim(tb18_class)
tb18_class[1:5,1:7]

#table with occurence data alone
tb18_tax_occur <- tb18_tax[,1:10]
dim(tb18_tax_occur) # 22702    10
tb18_tax_occur[1:5,1:10]

amplicons_per_sample_tb18<-colSums(tb18_tax_occur)
amplicons_per_sample_tb18[which(colSums(tb18_tax_occur)<11770)]
#there are no samples with less than 11770 reads. We'll normalize at 11770 reads.

#remove samples with less than 11770 reads
tb18_tax_occur_min11770 <- tb18_tax_occur[,colSums(tb18_tax_occur) >= 11770]
dim(tb18_tax_occur_min11770)

#remove samples with omitted in miTags dataset (so that we can compare their relative abundance cosidering the same samples):
tb18_tax_occur_min11770<-subset(tb18_tax_occur_min11770, select=-c(st030))
dim(tb18_tax_occur_min11770) #22702     9

#let's check if we loose any OTU when excluding the mentioned stations.
tb18_tax_occur_min11770_no_cero<-tb18_tax_occur_min11770[,-(which(colSums(tb18_tax_occur_min11770)==0))]
dim(tb18_tax_occur_min11770_no_cero)
```

Table dimensions and content outline:

```{r starting_dataset, echo=FALSE}
dim(tb18_tax_occur_min11770)
tb18_tax_occur_min11770[1:5,1:5]
```

Minimum number of reads per station:

```{r reads_per_sample_overview1, echo=1}
min(colSums(tb18_tax_occur_min11770)) 
#11770
```

Maximum number of reads per station:

```{r reads_per_sample_overview2, echo=1}
max(colSums(tb18_tax_occur_min11770)) 
# max: 275821
```

Identification of station with higher number of reads:

```{r reads_per_sample_overview3, echo=TRUE}
amplicons_per_sample<-colSums(tb18_tax_occur_min11770)
amplicons_per_sample[which(colSums(tb18_tax_occur_min11770)>275820)]
```

Overall reads per sample:

``` {r reads_per_sample_overview4, echo=FALSE}
plot(sort(colSums(tb18_tax_occur_min11770)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 1.2) Normalization

Let's normalize the original dataset by randomly subsampling 11770 reads in each station:

``` {r species_richness_rarefaction1, echo=TRUE}
library(vegan)
tb18_tax_occur_min11770_t<-t(tb18_tax_occur_min11770)
tb18_tax_occur_ss11770<-rrarefy(tb18_tax_occur_min11770_t, 11770)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2, echo=FALSE}
dim(tb18_tax_occur_ss11770)
tb18_tax_occur_ss11770[1:5,1:5]
```

Its content fits with the expected normalization values (11770 reads per station):

``` {r species_richness_rarefaction3, echo=TRUE}
rowSums(tb18_tax_occur_ss11770)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4, echo=1:5}
length(which(colSums(tb18_tax_occur_ss11770)==0)) 
```

There are 20487 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5, echo=1:3}
tb18_tax_occur_ss11770_no_cero<-tb18_tax_occur_ss11770[,-(which(colSums(tb18_tax_occur_ss11770)==0))]
dim(tb18_tax_occur_ss11770_no_cero)

#the final dimensions of the normalized table are 9 26657.
#26657 + 20487 = 47144
```

Datasets summary:
dim(tb18_tax) --> 22702    17
dim(tb18_tax_occur) --> 22702    10
dim(tb18_tax_occur_ss11770_no_cero) --> 9 8882





## 1.3) Taxonomic composition analysis

### 1.3.1) Normalized data

#### 1.3.1.1) Absolute values

Let's add the taxonomic classification to the left OTUs by merging "tb18_tax_occur_ss11770_no_cero" with "tb18_tax":

```{r merge_tables, echo=FALSE}
tb18_tax_occur_ss11770_no_cero_t<-t(tb18_tax_occur_ss11770_no_cero)
tb18_ss11770_tax<-merge(tb18_tax_occur_ss11770_no_cero_t,tb18_class, by="row.names")

dim(tb18_ss11770_tax)
tb18_ss11770_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_ss11770_tax)=tb18_ss11770_tax$Row.names

#add OTU_no as rowname
rownames.tb18_ss11770_tax<-tb18_ss11770_tax[,1]
tb18_ss11770_tax<-tb18_ss11770_tax[,-1]
#colnames(tb18_ss11770_tax, do.NULL=F)

dim(tb18_ss11770_tax)
tb18_ss11770_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_ss11770_tax["OTU_no"] <- NA
tb18_ss11770_tax$OTU_no <- sapply(strsplit(rownames(tb18_ss11770_tax),split= "\\_"),'[',2)
tb18_ss11770_tax$OTU_no <- as.numeric(as.character(tb18_ss11770_tax$OTU_no))
tb18_ss11770_tax_sorted<-tb18_ss11770_tax[order(tb18_ss11770_tax$OTU_no, decreasing = FALSE), ]

dim(tb18_ss11770_tax_sorted)
tb18_ss11770_tax_sorted[1:5,1:5]
```

```{r select_phototrophs, echo=FALSE}
# [TEMPORARY EDITION] for some reason, the taxonomic classification of "OTU_2485" is not properly parsed. As it is not classified as a phototroph anyway, by now let's get rid of it by excluding empty cells (""). 
tb18_phototrophs <- tb18_ss11770_tax_sorted[which(tb18_ss11770_tax_sorted$MAS_plus_BM_plus_SILVA_class != "" & tb18_ss11770_tax_sorted$MAS_plus_BM_plus_SILVA_class != "NA"),]

dim(tb18_phototrophs)
```

```{r count_samples_per_group, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:9]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
#length(Dinophyceae_tb_samples[which(colSums(Dinophyceae_tb_occur) != 0)])

Prasinophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Pinguiochysidales"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:9]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Eustigmatales"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:9]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:9]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:9]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:9]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- tb18_phototrophs[which(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:9]
write.table(Trebouxiophyceae_tb_occur, "Trebouxiophyceae_tb_occur.txt", sep = "\t")
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
```

```{r aggregate, echo=FALSE}
class_summary_reads_per_class<-aggregate(rowSums(tb18_phototrophs[1:9]), list(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class), sum)
# count the different groups

class_summary_otus_per_class<-aggregate(rowSums(tb18_phototrophs[1:9]), list(tb18_phototrophs$MAS_plus_BM_plus_SILVA_class), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class)
class_summary_reads_per_class_order<-class_summary_reads_per_class[order(-x),]
detach(class_summary_reads_per_class)
class_summary_reads_per_class_order

#fix column names
row.names(class_summary_reads_per_class_order)<-class_summary_reads_per_class_order[,1]
class_summary_reads_per_class_order<-class_summary_reads_per_class_order[c(-1)]
colnames(class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class)
class_summary_otus_per_class_order<-class_summary_otus_per_class[order(-x),]
detach(class_summary_otus_per_class)
class_summary_otus_per_class_order

row.names(class_summary_otus_per_class_order)<-class_summary_otus_per_class_order[,1]
class_summary_otus_per_class_order<-class_summary_otus_per_class_order[c(-1)]
colnames(class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples, echo=FALSE}
tb18S_OTUs_reads_samples <- merge(class_summary_reads_per_class_order, class_summary_otus_per_class_order, by="row.names")

row.names(tb18S_OTUs_reads_samples)<-tb18S_OTUs_reads_samples[,1]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[,-1]
colnames(tb18S_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
tb18S_OTUs_reads_samples[1:5,1:2]

tb18S_OTUs_reads_samples<-tb18S_OTUs_reads_samples[order(tb18S_OTUs_reads_samples$reads_per_class, tb18S_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples["samples_per_class"] <- NA
tb18S_OTUs_reads_samples$samples_per_class<-c(9,9,9,9,9,7,7,9,8,7,7,3,2,4,2,1)

tb18S_OTUs_reads_samples
```

OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs, echo=FALSE}
library(ggplot2)
library(ggrepel)

otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 12)) + labs(title="[1] OTUs per class vs. samples in which they occur", x="samples", y="OTUs") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_log10(limits = c(-1, 5000), breaks=c(0.1,1,10,100,1000,10000)) + scale_y_continuous(trans = "log10") + labs(title="[2] Reads per class vs. OTUs per class", x="OTUs (log10)", y="reads (log10)") 

ggplot(tb18S_OTUs_reads_samples) +
  geom_point(aes(OTUs_per_class,reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples)), size=4, force = 8, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_log10(limits = c(-1, 10000), breaks=c(0,1,10,100,1000,10000)) + scale_y_log10(limits = c(-1,100000), breaks = c(0,1,10,100,1000,10000,100000)) + labs(title="[2.2] Reads per class vs. OTUs per class", x="\nMP-iTags-fraction-18S OTUs", y="MP-iTags-fraction-18S reads\n") + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

#### 1.3.1.1) Relative values


```{r OTUs_vs_reads_relative_values_18S, echo=FALSE}

tb18S_OTUs_reads_samples_rel_abund <- tb18S_OTUs_reads_samples

tb18S_OTUs_reads_samples_rel_abund$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples)[1]
tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples)[2]
tb18S_OTUs_reads_samples_rel_abund$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund$samples_per_class*100)/9

colSums(tb18S_OTUs_reads_samples_rel_abund)
tb18S_OTUs_reads_samples_rel_abund
```

OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund, aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[3] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")


ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,OTUs_per_class)) +
  geom_text_repel(aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund)), size=3, force = 3, box.padding = unit(0.20, "lines"), point.padding = unit(0.25, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[3.2] OTUs per class vs. samples in which they occur", x="MP-iTags-fraction-18S samples (%)", y="MP-iTags-fraction-18S OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + labs(title="[4] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1,200), breaks = c(0.1,1,10,100))


ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund)), size=3, force = 3, box.padding = unit(0.20, "lines"), point.padding = unit(0.25, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[4.2] Reads per class vs. OTUs per class", x="MP-iTags-fraction-18S OTUs (%)", y="MP-iTags-fraction-18S reads (%)") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1,200), breaks = c(0.1,1,10,100))

ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[4.3] Reads per class vs. OTUs per class", x="\nMP-iTags-fraction-18S OTUs (%)", y="MP-iTags-fraction-18S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

```{r reads_vs_samples_18S, echo=FALSE}
ggplot(tb18S_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund)), size=4, force = 4, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[New] Reads per class vs. samples in which they occur", x="\nMP-iTags-fraction-18S samples (%)", y="MP-iTags-fraction-18S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


### 1.3.2) Non-normalized data

```{r merge_tables_non.norm_18S, echo=FALSE}
tb18_tax_occur_min11770_t<-tb18_tax_occur_min11770
tb18_min11770_tax<-merge(tb18_tax_occur_min11770_t,tb18_class, by="row.names")

dim(tb18_min11770_tax)
tb18_min11770_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb18_min11770_tax)=tb18_min11770_tax$Row.names

#add OTU_no as rowname
rownames.tb18_min11770_tax<-tb18_min11770_tax[,1]
tb18_min11770_tax<-tb18_min11770_tax[,-1]
#colnames(tb18_min11770_tax, do.NULL=F)

dim(tb18_min11770_tax)
tb18_min11770_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb18_min11770_tax["OTU_no"] <- NA
tb18_min11770_tax$OTU_no <- sapply(strsplit(rownames(tb18_min11770_tax),split= "\\_"),'[',2)
tb18_min11770_tax$OTU_no <- as.numeric(as.character(tb18_min11770_tax$OTU_no))
tb18_min11770_tax_sorted<-tb18_min11770_tax[order(tb18_min11770_tax$OTU_no, decreasing = FALSE), ]

dim(tb18_min11770_tax_sorted)
tb18_min11770_tax_sorted[1:5,1:5]
```

```{r select_phototrophs_non.norm_18S, echo=FALSE}
tb18_phototrophs_non.norm <- tb18_min11770_tax_sorted[which(tb18_min11770_tax_sorted$MAS_plus_BM_plus_SILVA_class != "" & tb18_min11770_tax_sorted$MAS_plus_BM_plus_SILVA_class != "NA"),]

dim(tb18_phototrophs_non.norm)
```

```{r count_samples_per_group_non.norm_18S, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:9]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])

Prasinophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Pinguiochysidales"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:9]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Eustigmatales"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:9]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:9]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:9]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:9]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- tb18_phototrophs_non.norm[which(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:9]
write.table(Trebouxiophyceae_tb_occur, "Trebouxiophyceae_tb_occur.txt", sep = "\t")
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])
```

```{r aggregate_non.norm_18S, echo=FALSE}
class_summary_reads_per_class_non.norm<-aggregate(rowSums(tb18_phototrophs_non.norm[1:9]), list(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class), sum)
# count the different groups

class_summary_otus_per_class_non.norm<-aggregate(rowSums(tb18_phototrophs_non.norm[1:9]), list(tb18_phototrophs_non.norm$MAS_plus_BM_plus_SILVA_class), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_non.norm)
class_summary_reads_per_class_non.norm_order<-class_summary_reads_per_class_non.norm[order(-x),]
detach(class_summary_reads_per_class_non.norm)
class_summary_reads_per_class_non.norm_order

#fix column names
row.names(class_summary_reads_per_class_non.norm_order)<-class_summary_reads_per_class_non.norm_order[,1]
class_summary_reads_per_class_non.norm_order<-class_summary_reads_per_class_non.norm_order[c(-1)]
colnames(class_summary_reads_per_class_non.norm_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(class_summary_otus_per_class_non.norm)
class_summary_otus_per_class_non.norm_order<-class_summary_otus_per_class_non.norm[order(-x),]
detach(class_summary_otus_per_class_non.norm)
class_summary_otus_per_class_non.norm_order

row.names(class_summary_otus_per_class_non.norm_order)<-class_summary_otus_per_class_non.norm_order[,1]
class_summary_otus_per_class_non.norm_order<-class_summary_otus_per_class_non.norm_order[c(-1)]
colnames(class_summary_otus_per_class_non.norm_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_non.norm_18S, echo=FALSE}
tb18S_OTUs_reads_samples_non.norm <- merge(class_summary_reads_per_class_non.norm_order, class_summary_otus_per_class_non.norm_order, by="row.names")

row.names(tb18S_OTUs_reads_samples_non.norm)<-tb18S_OTUs_reads_samples_non.norm[,1]

tb18S_OTUs_reads_samples_non.norm<-tb18S_OTUs_reads_samples_non.norm[,-1]
colnames(tb18S_OTUs_reads_samples_non.norm)<-c("reads_per_class","OTUs_per_class")
tb18S_OTUs_reads_samples_non.norm[1:5,1:2]

tb18S_OTUs_reads_samples_non.norm<-tb18S_OTUs_reads_samples_non.norm[order(tb18S_OTUs_reads_samples_non.norm$reads_per_class, tb18S_OTUs_reads_samples_non.norm$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb18S_OTUs_reads_samples_non.norm["samples_per_class"] <- NA
tb18S_OTUs_reads_samples_non.norm$samples_per_class<-c(9,9,9,9,9,8,9,9,9,9,9,2,4,8,5,6,1)

tb18S_OTUs_reads_samples_non.norm
```

```{r OTUs_vs_reads_relative_values_non.norm_18S, echo=FALSE}
tb18S_OTUs_reads_samples_rel_abund_non.norm <- tb18S_OTUs_reads_samples_non.norm

tb18S_OTUs_reads_samples_rel_abund_non.norm$reads_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$reads_per_class*100)/colSums(tb18S_OTUs_reads_samples_non.norm)[1]
tb18S_OTUs_reads_samples_rel_abund_non.norm$OTUs_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$OTUs_per_class*100)/colSums(tb18S_OTUs_reads_samples_non.norm)[2]
tb18S_OTUs_reads_samples_rel_abund_non.norm$samples_per_class<-(tb18S_OTUs_reads_samples_rel_abund_non.norm$samples_per_class*100)/9

colSums(tb18S_OTUs_reads_samples_rel_abund_non.norm)
tb18S_OTUs_reads_samples_rel_abund_non.norm
```

OTUs per class vs. samples in which they occur [PLOT DESCRIPTION]

```{r stations_vs_OTUs_non.norm_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund_non.norm, aes(samples_per_class,OTUs_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund_non.norm)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 118), breaks=c(25,50,75,100)) + labs(title="[5] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class [PLOT DESCRIPTION]

```{r OTUs_vs_reads_non.norm_18S, echo=FALSE}
otus_vs_samples_plot <- ggplot(tb18S_OTUs_reads_samples_rel_abund_non.norm, aes(OTUs_per_class, reads_per_class, label = rownames(tb18S_OTUs_reads_samples_rel_abund_non.norm)))
otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + labs(title="[6] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") + scale_x_log10(limits = c(-1, 200), breaks=c(0.1,1,10,100))
```





# 2) 16S amplicons

## 2.1) Data overview

Let's read the dataset and remove the samples containing less than 5836 reads:

``` {r load_data_16S, echo=FALSE, message=FALSE}
#read data 
tb16_tax <- read.table(file="/home/laura/Documents/TFM/home/data/MALASPINA/Malaspina_16S_Surface/amplicons_fraction/MP_16S_SRF_tax_classif_tab_filtered_v2_amplicons_fraction_sorted.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(tb16_tax) # 753  17
tb16_tax[1:5,1:5]

#row names = OTU name (option A)
row.names(tb16_tax)<-tb16_tax[,1]

#row names = row number (option B)
#rownames(otu_tb16) <- 1:nrow(otu_tb16)

tb16_tax<-tb16_tax[,-1]
tb16_tax[is.na(tb16_tax)]<-0

dim(tb16_tax)
tb16_tax[1:5,1:5]

#table with taxonomi classification alone
tb16_class <- tb16_tax[,11:16]
dim(tb16_class)
tb16_class[1:5,1:6]

#table with occurence data alone
tb16_tax_occur <- tb16_tax[,1:10]
dim(tb16_tax_occur) # 753  10
tb16_tax_occur[1:5,1:5]

amplicons_per_sample_tb16<-colSums(tb16_tax_occur)
amplicons_per_sample_tb16[which(colSums(tb16_tax_occur)<5836)]
#No samples have less than 5836 reads. We'll normalize at 5836 reads.

#remove samples with less than 5836 reads
tb16_tax_occur_min5836 <- tb16_tax_occur[,colSums(tb16_tax_occur) >= 5836]
dim(tb16_tax_occur_min5836)

#remove samples omitted in miTags dataset (so that we can compare the relative abundance of their OTUs considering the same samples):
tb16_tax_occur_min5836<-subset(tb16_tax_occur_min5836, select=-c(st030))
dim(tb16_tax_occur_min5836)


#let's check if we loose any OTU when excluding the mentioned stations.
tb16_tax_occur_min5836_no_cero<-tb16_tax_occur_min5836[,-(which(colSums(tb16_tax_occur_min5836)==0))]
dim(tb16_tax_occur_min5836_no_cero)
```


Table dimensions and content outline:

```{r starting_dataset_16S, echo=FALSE}
dim(tb16_tax_occur_min5836)
tb16_tax_occur_min5836[1:5,1:5]
```

Minimum number of reads per station:

```{r reads_per_sample_overview1_16S, echo=1}
min(colSums(tb16_tax_occur_min5836)) 
#5836
```

Maximum number of reads per station:

```{r reads_per_sample_overview2_16S, echo=1}
max(colSums(tb16_tax_occur_min5836)) 
# max: 97929
```

Identification of station with higher number of reads:

```{r reads_per_sample_overview3_16S, echo=TRUE}
amplicons_per_sample<-colSums(tb16_tax_occur_min5836)
amplicons_per_sample[which(colSums(tb16_tax_occur_min5836)>97928)]
```

Overall reads per sample:

``` {r reads_per_sample_overview4_16S, echo=FALSE}
plot(sort(colSums(tb16_tax_occur_min5836)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 2.2) Normalization

Let's normalize the original dataset by randomly subsampling 5836 reads in each station:

``` {r species_richness_rarefaction1_16S, echo=TRUE}
library(vegan)
tb16_tax_occur_min5836_t<-t(tb16_tax_occur_min5836)
tb16_tax_occur_ss5836<-rrarefy(tb16_tax_occur_min5836_t, 5836)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2_16S, echo=FALSE}
dim(tb16_tax_occur_ss5836)
tb16_tax_occur_ss5836[1:5,1:5]
```

Its content fits with the expected normalization values (5836 reads per station):

``` {r species_richness_rarefaction3_16S, echo=TRUE}
rowSums(tb16_tax_occur_ss5836)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4_16S, echo=1:5}
length(which(colSums(tb16_tax_occur_ss5836)==0)) 
```

There are 54 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5_16S, echo=1:3}
tb16_tax_occur_ss5836_no_cero<-tb16_tax_occur_ss5836[,-(which(colSums(tb16_tax_occur_ss5836)==0))]
dim(tb16_tax_occur_ss5836_no_cero)

#the final dimensions of the normalized table are 123 27067.
#939 + 54 = 993
```

Datasets summary:
dim(tb16_tax) --> 753  16
dim(tb16_tax_occur) --> 753  10
dim(tb16_tax_occur_ss5836_no_cero) --> 9 575



## 2.3) Taxonomic composition analysis

### 2.3.1) Normalized data

#### 2.3.1.1) Absolute values

Let's add the taxonomic classification to the left OTUs by merging "tb16_tax_occur_ss5836_no_cero" with "tb16_tax".

```{r merge_tables_16S, echo=FALSE}
tb16_tax_occur_ss5836_no_cero_t<-t(tb16_tax_occur_ss5836_no_cero)
tb16_ss5836_tax<-merge(tb16_tax_occur_ss5836_no_cero_t,tb16_class, by="row.names")

dim(tb16_ss5836_tax)
tb16_ss5836_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb16_ss5836_tax)=tb16_ss5836_tax$Row.names

#add OTU_no as rowname
rownames.tb16_ss5836_tax<-tb16_ss5836_tax[,1]
tb16_ss5836_tax<-tb16_ss5836_tax[,-1]
#colnames(tb16_ss5836_tax, do.NULL=F)

dim(tb16_ss5836_tax)
tb16_ss5836_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb16_ss5836_tax["OTU_no"] <- NA
tb16_ss5836_tax$OTU_no <- sapply(strsplit(rownames(tb16_ss5836_tax),split= "\\_"),'[',2)
tb16_ss5836_tax$OTU_no <- as.numeric(as.character(tb16_ss5836_tax$OTU_no))
tb16_ss5836_tax_sorted<-tb16_ss5836_tax[order(tb16_ss5836_tax$OTU_no, decreasing = FALSE), ]

dim(tb16_ss5836_tax_sorted)
tb16_ss5836_tax_sorted[1:5,14:16]
```

```{r select_phototrophs_16S2, echo=FALSE}
tb16_bacteria <- tb16_ss5836_tax_sorted[which(tb16_ss5836_tax_sorted$class_A != "Bolidophyceae" & tb16_ss5836_tax_sorted$class_A != "Chlorarachniophyta" & tb16_ss5836_tax_sorted$class_A != "Cryptophyceae" & tb16_ss5836_tax_sorted$class_A != "Pelagophyceae" & tb16_ss5836_tax_sorted$class_A != "Rappemonad" & tb16_ss5836_tax_sorted$class_A != "Cryptomonadales" & tb16_ss5836_tax_sorted$class_A != "Bacillariophyta" & tb16_ss5836_tax_sorted$class_A != "Dictyochophyceae" & tb16_ss5836_tax_sorted$class_A != "Mamiellophyceae" & tb16_ss5836_tax_sorted$class_A != "Prasinophyceae" & tb16_ss5836_tax_sorted$class_A != "Eustigmatales" & tb16_ss5836_tax_sorted$class_A != "Trebouxiophyceae" & tb16_ss5836_tax_sorted$class_A != "Prymnesiophyceae" & tb16_ss5836_tax_sorted$class_A != "other_plastids" & tb16_ss5836_tax_sorted$class_A != "Chrysophyceae" ),]

tb16_protists <- tb16_ss5836_tax_sorted[which(tb16_ss5836_tax_sorted$class_A != "heterotrophic_bacteria" &  tb16_ss5836_tax_sorted$class_A != "Cyanobacteria"),]

dim(tb16_protists)
dim(tb16_bacteria)
tb16_protists[1:5,10:16]
tb16_bacteria[1:5,10:16]
```

```{r aggregate_16S_protists, echo=TRUE}
class_summary_reads_per_class_16S_protists<-aggregate(rowSums(tb16_protists[1:9]), list(tb16_protists$class_A), sum)
# count the different groups

class_summary_otus_per_class_16S_protists<-aggregate(rowSums(tb16_protists[1:9]), list(tb16_protists$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_16S_protists)
class_summary_reads_per_class_16S_protists_order<-class_summary_reads_per_class_16S_protists[order(-x),]
detach(class_summary_reads_per_class_16S_protists)
class_summary_reads_per_class_16S_protists_order

#fix column names
row.names(class_summary_reads_per_class_16S_protists_order)<-class_summary_reads_per_class_16S_protists_order[,1]
class_summary_reads_per_class_16S_protists_order<-class_summary_reads_per_class_16S_protists_order[c(-1)]
colnames(class_summary_reads_per_class_16S_protists_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_16S_protists)
class_summary_otus_per_class_16S_protists_order<-class_summary_otus_per_class_16S_protists[order(-x),]
detach(class_summary_otus_per_class_16S_protists)
class_summary_otus_per_class_16S_protists_order

row.names(class_summary_otus_per_class_16S_protists_order)<-class_summary_otus_per_class_16S_protists_order[,1]
class_summary_otus_per_class_16S_protists_order<-class_summary_otus_per_class_16S_protists_order[c(-1)]
colnames(class_summary_otus_per_class_16S_protists_order)<-c("OTUs_per_class")
```

```{r aggregate_16S_bacteria, echo=TRUE}
class_summary_reads_per_class_16S_bacteria<-aggregate(rowSums(tb16_bacteria[1:9]), list(tb16_bacteria$class_A), sum)
# count the different groups

class_summary_otus_per_class_16S_bacteria<-aggregate(rowSums(tb16_bacteria[1:9]), list(tb16_bacteria$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_16S_bacteria)
class_summary_reads_per_class_16S_bacteria_order<-class_summary_reads_per_class_16S_bacteria[order(-x),]
detach(class_summary_reads_per_class_16S_bacteria)
class_summary_reads_per_class_16S_bacteria_order

#fix column names
row.names(class_summary_reads_per_class_16S_bacteria_order)<-class_summary_reads_per_class_16S_bacteria_order[,1]
class_summary_reads_per_class_16S_bacteria_order<-class_summary_reads_per_class_16S_bacteria_order[c(-1)]
colnames(class_summary_reads_per_class_16S_bacteria_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_16S_bacteria)
class_summary_otus_per_class_16S_bacteria_order<-class_summary_otus_per_class_16S_bacteria[order(-x),]
detach(class_summary_otus_per_class_16S_bacteria)
class_summary_otus_per_class_16S_bacteria_order

row.names(class_summary_otus_per_class_16S_bacteria_order)<-class_summary_otus_per_class_16S_bacteria_order[,1]
class_summary_otus_per_class_16S_bacteria_order<-class_summary_otus_per_class_16S_bacteria_order[c(-1)]
colnames(class_summary_otus_per_class_16S_bacteria_order)<-c("OTUs_per_class")
```

```{r count_samples_per_group_16S_protists, echo=FALSE}
#create a table per group and count in how many samples they occur. 
other_plastids_tb <- tb16_protists[which(tb16_protists$class_A == "other_plastids"),]
other_plastids_tb[1:5,1:5]
other_plastids_tb_occur <- other_plastids_tb[,1:9]
other_plastids_tb_occur[1:5,1:5]
dim(other_plastids_tb_occur)
length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])

Prymnesiophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Prasinophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Bacillariophyta_tb <- tb16_protists[which(tb16_protists$class_A == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Chlorarachniophyta_tb <- tb16_protists[which(tb16_protists$class_A == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Cryptomonadales_tb <- tb16_protists[which(tb16_protists$class_A == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Pelagophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Rappemonad_tb <- tb16_protists[which(tb16_protists$class_A == "Rappemonad"),]
Rappemonad_tb_occur <- Rappemonad_tb[,1:9]
length(Rappemonad_tb_occur[,colSums(Rappemonad_tb_occur) > 0])

Chrysophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
```

```{r count_samples_per_group_16S_bacteria, echo=FALSE}
#create a table per group and count in how many samples they occur. 
heterotrophic_bacteria_tb <- tb16_bacteria[which(tb16_bacteria$class_A == "heterotrophic_bacteria"),]
heterotrophic_bacteria_tb_occur <- heterotrophic_bacteria_tb[,1:9]
length(heterotrophic_bacteria_tb_occur[,colSums(heterotrophic_bacteria_tb_occur) > 0])

Cyanobacteria_tb <- tb16_bacteria[which(tb16_bacteria$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:9]
length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
```

```{r create_table_merging_#OTUs_#reads_#samples_protists, echo=FALSE}
tb16S_protists_OTUs_reads_samples <- merge(class_summary_reads_per_class_16S_protists_order, class_summary_otus_per_class_16S_protists_order, by="row.names")

row.names(tb16S_protists_OTUs_reads_samples)<-tb16S_protists_OTUs_reads_samples[,1]
tb16S_protists_OTUs_reads_samples<-tb16S_protists_OTUs_reads_samples[,-1]
colnames(tb16S_protists_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
tb16S_protists_OTUs_reads_samples[1:5,1:2]

tb16S_protists_OTUs_reads_samples<-tb16S_protists_OTUs_reads_samples[order(tb16S_protists_OTUs_reads_samples$reads_per_class, tb16S_protists_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_protists_OTUs_reads_samples["samples_per_class"] <- NA
tb16S_protists_OTUs_reads_samples$samples_per_class<-c(9,9,7,9,2,1,1,1,1,2,1,1)

#table without "other_plastids"
tb16S_protists_OTUs_reads_samples <- tb16S_protists_OTUs_reads_samples[-2,]
```

```{r create_table_merging_#OTUs_#reads_#samples_bacteria, echo=FALSE}
tb16S_bacteria_OTUs_reads_samples <- merge(class_summary_reads_per_class_16S_bacteria_order, class_summary_otus_per_class_16S_bacteria_order, by="row.names")

row.names(tb16S_bacteria_OTUs_reads_samples)<-tb16S_bacteria_OTUs_reads_samples[,1]
tb16S_bacteria_OTUs_reads_samples<-tb16S_bacteria_OTUs_reads_samples[,-1]
colnames(tb16S_bacteria_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
tb16S_bacteria_OTUs_reads_samples[1:5,1:2]

tb16S_bacteria_OTUs_reads_samples<-tb16S_bacteria_OTUs_reads_samples[order(tb16S_bacteria_OTUs_reads_samples$reads_per_class, tb16S_bacteria_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_bacteria_OTUs_reads_samples["samples_per_class"] <- NA
tb16S_bacteria_OTUs_reads_samples$samples_per_class<-c(9,9)

#table without "heterotropic_bacteria":
tb16S_bacteria_OTUs_reads_samples <- tb16S_bacteria_OTUs_reads_samples[-1,]
```


OTUs per class vs. samples in which they occur - PROTISTS [PLOT DESCRIPTION]

```{r stations_vs_OTUs_protists, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_OTUs_reads_samples)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 11)) + labs(title="[7] OTUs per class vs. samples in which they occur", x="samples", y="OTUs") 
#coord_trans(y = "log10")
```

Reads per class vs. OTUs per class - PROTISTS [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_OTUs_reads_samples)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 26)) + labs(title="[8] Reads per class vs. OTUs per class", x="OTUs", y="reads") + coord_trans(y = "log10")
```

OTUs per class vs. samples in which they occur - PROTISTS vs CYANOBACTERIA [PLOT DESCRIPTION]

```{r merge_protists_and_bacteria_tb, echo=FALSE}
tb16S_protists_plus_bacteria_OTUs_reads_samples<-rbind(tb16S_protists_OTUs_reads_samples,tb16S_bacteria_OTUs_reads_samples)
```

```{r stations_vs_OTUs_protists_plus_bacteria, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 11)) + labs(title="[9] OTUs per class vs. samples in which they occur", x="samples", y="OTUs")
```

Reads per class vs. OTUs per class - PROTISTS vs CYANOBACTERIA [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_plus_bacteria, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 26)) + labs(title="[10] Reads per class vs. OTUs per class", x="OTUs", y="reads") + coord_trans(y = "log10")

ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples) +
  geom_point(aes(OTUs_per_class,reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class,reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples)), size=4, force = 3, box.padding = unit(0.15, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_log10(limits = c(-1,100), breaks=c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,100000), breaks = c(0,0.1,1,10,100,1000,10000,100000)) + labs(title="[10.2] Reads per class vs. OTUs per class", x="\nMP-iTags-fraction-16S OTUs", y="MP-iTags-fraction-16S reads\n") + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

#### 2.3.1.2) Relative values

```{r OTUs_vs_reads_protists_relative_values, echo=FALSE}

tb16S_protists_OTUs_reads_samples_rel_abund <- tb16S_protists_OTUs_reads_samples

tb16S_protists_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_protists_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_protists_OTUs_reads_samples)[1]
tb16S_protists_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_protists_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_protists_OTUs_reads_samples)[2]
tb16S_protists_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_protists_OTUs_reads_samples_rel_abund$samples_per_class*100)/9

colSums(tb16S_protists_OTUs_reads_samples_rel_abund)
tb16S_protists_OTUs_reads_samples_rel_abund
```

OTUs per class vs. samples in which they occur - PROTISTS [PLOT DESCRIPTION]

```{r stations_vs_OTUs_protists_16S, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples_rel_abund, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_OTUs_reads_samples)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[11] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class - PROTISTS [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_16S, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_OTUs_reads_samples)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 60)) + labs(title="[12] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10")
```

OTUs per class vs. samples in which they occur - PROTISTS vs. CYANOBACTERIA [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_plus_bacteria_relative_values, echo=FALSE}

tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund<-tb16S_protists_plus_bacteria_OTUs_reads_samples

tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$reads_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$reads_per_class*100)/colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples)[1]

tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$OTUs_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$OTUs_per_class*100)/colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples)[2]

tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$samples_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund$samples_per_class*100)/9

colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)
tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund
```

```{r stations_vs_OTUs_protists_plus_bacteria_rel_abundance, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[13] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")

ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,OTUs_per_class)) +
  geom_text_repel(aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)), size=3, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.1, 'lines'), nudge_y = 2, nudge_x = 0.5) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[13.2] OTUs per class vs. samples in which they occur", x="MP-iTags-fraction-16S samples (%)", y="MP-iTags-fraction-16S OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class - PROTISTS vs CYANOBACTERIA [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_plus_bacteria_rel_abundance, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 42)) + labs(title="[14] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") 


ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)), size=3, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.1, 'lines'), nudge_y = 0.005, nudge_x = 0.005) + scale_x_continuous(limits = c(0, 42)) + labs(title="[14.2] Reads per class vs. OTUs per class", x="MP-iTags-fraction-16S OTUs (%)", y="MP-iTags-fraction-16S reads (%)") + coord_trans(y = "log10") 

ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund) +
  geom_point(aes(OTUs_per_class, reads_per_class)) +
  geom_text_repel(aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)), size=4, force = 3, box.padding = unit(0.1, "lines"), point.padding = unit(0.02, 'lines')) + labs(title="[14.3] Reads per class vs. OTUs per class", x="\nMP-iTags-fraction-16S OTUs (%)", y="MP-iTags-fraction-16S reads (%)\n") + scale_x_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100))  + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```

```{r reads_vs_samples_16S, echo=FALSE}
ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund) +
  geom_point(aes(samples_per_class,reads_per_class)) +
  geom_text_repel(aes(samples_per_class,reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_rel_abund)), size=4, force = 5, box.padding = unit(0.1, "lines"), point.padding = unit(0.2, 'lines')) + scale_x_continuous(limits = c(0, 116), breaks=c(25,50,75,100)) + labs(title="[New] Reads per class vs. samples in which they occur", x="\nMP-iTags-fraction-16S samples (%)", y="MP-iTags-fraction-16S reads (%)\n") + scale_y_log10(limits = c(-1,200), breaks = c(0,0.1,1,10,100)) + theme(axis.text=element_text(size=14), axis.title=element_text(size=16))
```


### 2.3.2) Non-normalized data

```{r merge_tables_16S_non.norm, echo=FALSE}
tb16_tax_occur_min5836_no_cero_t<-tb16_tax_occur_min5836
tb16_min5836_tax<-merge(tb16_tax_occur_min5836_no_cero_t,tb16_class, by="row.names")

dim(tb16_min5836_tax)
tb16_min5836_tax[1:5,1:5]

#fix OTU_no as new row
rownames(tb16_min5836_tax)=tb16_min5836_tax$Row.names

#add OTU_no as rowname
rownames.tb16_min5836_tax<-tb16_min5836_tax[,1]
tb16_min5836_tax<-tb16_min5836_tax[,-1]
#colnames(tb16_min5836_tax, do.NULL=F)

dim(tb16_min5836_tax)
tb16_min5836_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
tb16_min5836_tax["OTU_no"] <- NA
tb16_min5836_tax$OTU_no <- sapply(strsplit(rownames(tb16_min5836_tax),split= "\\_"),'[',2)
tb16_min5836_tax$OTU_no <- as.numeric(as.character(tb16_min5836_tax$OTU_no))
tb16_min5836_tax_sorted<-tb16_min5836_tax[order(tb16_min5836_tax$OTU_no, decreasing = FALSE), ]

dim(tb16_min5836_tax_sorted)
tb16_min5836_tax_sorted[1:5,14:16]
```

```{r select_phototrophs_16S_non.norm, echo=FALSE}
tb16_bacteria_non.norm <- tb16_min5836_tax_sorted[which(tb16_min5836_tax_sorted$class_A != "Bolidophyceae" & tb16_min5836_tax_sorted$class_A != "Chlorarachniophyta" & tb16_min5836_tax_sorted$class_A != "Cryptophyceae" & tb16_min5836_tax_sorted$class_A != "Pelagophyceae" & tb16_min5836_tax_sorted$class_A != "Rappemonad" & tb16_min5836_tax_sorted$class_A != "Cryptomonadales" & tb16_min5836_tax_sorted$class_A != "Bacillariophyta" & tb16_min5836_tax_sorted$class_A != "Dictyochophyceae" & tb16_min5836_tax_sorted$class_A != "Mamiellophyceae" & tb16_min5836_tax_sorted$class_A != "Prasinophyceae" & tb16_min5836_tax_sorted$class_A != "Eustigmatales" & tb16_min5836_tax_sorted$class_A != "Trebouxiophyceae" & tb16_min5836_tax_sorted$class_A != "Prymnesiophyceae" & tb16_min5836_tax_sorted$class_A != "Chrysophyceae" & tb16_min5836_tax_sorted$class_A != "other_plastids"),]

tb16_protists_non.norm <- tb16_min5836_tax_sorted[which(tb16_min5836_tax_sorted$class_A != "heterotrophic_bacteria" & tb16_min5836_tax_sorted$class_A != "Cyanobacteria"),]

dim(tb16_protists_non.norm)
dim(tb16_bacteria_non.norm)
tb16_protists_non.norm[1:5,]
tb16_bacteria_non.norm[1:5,]
```

```{r aggregate_16S_protists_non.norm, echo=TRUE}
class_summary_reads_per_class_16S_protists_non.norm<-aggregate(rowSums(tb16_protists_non.norm[1:9]), list(tb16_protists_non.norm$class_A), sum)
# count the different groups

class_summary_otus_per_class_16S_protists_non.norm<-aggregate(rowSums(tb16_protists_non.norm[1:9]), list(tb16_protists_non.norm$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_16S_protists_non.norm)
class_summary_reads_per_class_16S_protists_non.norm_order<-class_summary_reads_per_class_16S_protists_non.norm[order(-x),]
detach(class_summary_reads_per_class_16S_protists_non.norm)
class_summary_reads_per_class_16S_protists_non.norm_order

#fix column names
row.names(class_summary_reads_per_class_16S_protists_non.norm_order)<-class_summary_reads_per_class_16S_protists_non.norm_order[,1]
class_summary_reads_per_class_16S_protists_non.norm_order<-class_summary_reads_per_class_16S_protists_non.norm_order[c(-1)]
colnames(class_summary_reads_per_class_16S_protists_non.norm_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_16S_protists_non.norm)
class_summary_otus_per_class_16S_protists_non.norm_order<-class_summary_otus_per_class_16S_protists_non.norm[order(-x),]
detach(class_summary_otus_per_class_16S_protists_non.norm)
class_summary_otus_per_class_16S_protists_non.norm_order

row.names(class_summary_otus_per_class_16S_protists_non.norm_order)<-class_summary_otus_per_class_16S_protists_non.norm_order[,1]
class_summary_otus_per_class_16S_protists_non.norm_order<-class_summary_otus_per_class_16S_protists_non.norm_order[c(-1)]
colnames(class_summary_otus_per_class_16S_protists_non.norm_order)<-c("OTUs_per_class")
```

```{r aggregate_16S_bacteria_non.norm, echo=TRUE}
class_summary_reads_per_class_16S_bacteria_non.norm<-aggregate(rowSums(tb16_bacteria_non.norm[1:9]), list(tb16_bacteria_non.norm$class_A), sum)
# count the different groups

class_summary_otus_per_class_16S_bacteria_non.norm<-aggregate(rowSums(tb16_bacteria_non.norm[1:9]), list(tb16_bacteria_non.norm$class_A), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_16S_bacteria_non.norm)
class_summary_reads_per_class_16S_bacteria_non.norm_order<-class_summary_reads_per_class_16S_bacteria_non.norm[order(-x),]
detach(class_summary_reads_per_class_16S_bacteria_non.norm)
class_summary_reads_per_class_16S_bacteria_non.norm_order

#fix column names
row.names(class_summary_reads_per_class_16S_bacteria_non.norm_order)<-class_summary_reads_per_class_16S_bacteria_non.norm_order[,1]
class_summary_reads_per_class_16S_bacteria_non.norm_order<-class_summary_reads_per_class_16S_bacteria_non.norm_order[c(-1)]
colnames(class_summary_reads_per_class_16S_bacteria_non.norm_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_16S_bacteria_non.norm)
class_summary_otus_per_class_16S_bacteria_non.norm_order<-class_summary_otus_per_class_16S_bacteria_non.norm[order(-x),]
detach(class_summary_otus_per_class_16S_bacteria_non.norm)
class_summary_otus_per_class_16S_bacteria_non.norm_order

row.names(class_summary_otus_per_class_16S_bacteria_non.norm_order)<-class_summary_otus_per_class_16S_bacteria_non.norm_order[,1]
class_summary_otus_per_class_16S_bacteria_non.norm_order<-class_summary_otus_per_class_16S_bacteria_non.norm_order[c(-1)]
colnames(class_summary_otus_per_class_16S_bacteria_non.norm_order)<-c("OTUs_per_class")
```

```{r count_samples_per_group_16S_protists_non.norm, echo=FALSE}
#create a table per group and count in how many samples they occur. 
other_plastids_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "other_plastids"),]
other_plastids_tb[1:5,1:5]
other_plastids_tb_occur <- other_plastids_tb[,1:9]
other_plastids_tb_occur[1:5,1:5]
dim(other_plastids_tb_occur)
length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])

Prymnesiophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Prasinophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Bacillariophyta_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Bolidophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Chlorarachniophyta_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Cryptomonadales_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Pelagophyceae_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Rappemonad_tb <- tb16_protists_non.norm[which(tb16_protists_non.norm$class_A == "Rappemonad"),]
Rappemonad_tb_occur <- Rappemonad_tb[,1:9]
length(Rappemonad_tb_occur[,colSums(Rappemonad_tb_occur) > 0])

Chrysophyceae_tb <- tb16_protists[which(tb16_protists$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])
```

[[BACTERIA]]

```{r count_samples_per_group_16S_bacteria_non.norm, echo=FALSE}
#create a table per group and count in how many samples they occur. 
heterotrophic_bacteria_tb <- tb16_bacteria_non.norm[which(tb16_bacteria_non.norm$class_A == "heterotrophic_bacteria"),]
heterotrophic_bacteria_tb_occur <- heterotrophic_bacteria_tb[,1:9]
length(heterotrophic_bacteria_tb_occur[,colSums(heterotrophic_bacteria_tb_occur) > 0])

Cyanobacteria_tb <- tb16_bacteria_non.norm[which(tb16_bacteria_non.norm$class_A == "Cyanobacteria"),]
Cyanobacteria_tb_occur <- Cyanobacteria_tb[,1:9]
length(Cyanobacteria_tb_occur[,colSums(Cyanobacteria_tb_occur) > 0])
```

```{r create_table_merging_#OTUs_#reads_#samples_protists_non.norm, echo=FALSE}
tb16S_protists_OTUs_reads_samples_non.norm <- merge(class_summary_reads_per_class_16S_protists_non.norm_order, class_summary_otus_per_class_16S_protists_non.norm_order, by="row.names")

row.names(tb16S_protists_OTUs_reads_samples_non.norm)<-tb16S_protists_OTUs_reads_samples_non.norm[,1]
tb16S_protists_OTUs_reads_samples_non.norm<-tb16S_protists_OTUs_reads_samples_non.norm[,-1]
colnames(tb16S_protists_OTUs_reads_samples_non.norm)<-c("reads_per_class","OTUs_per_class")
tb16S_protists_OTUs_reads_samples_non.norm[1:5,1:2]

tb16S_protists_OTUs_reads_samples_non.norm<-tb16S_protists_OTUs_reads_samples_non.norm[order(tb16S_protists_OTUs_reads_samples_non.norm$reads_per_class, tb16S_protists_OTUs_reads_samples_non.norm$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_protists_OTUs_reads_samples_non.norm["samples_per_class"] <- NA
tb16S_protists_OTUs_reads_samples_non.norm$samples_per_class<-c(9,9,9,9,4,4,4,2,1,4,1,1)

#table without "other_plastids"
tb16S_protists_OTUs_reads_samples_non.norm <- tb16S_protists_OTUs_reads_samples_non.norm[-2,]
```

```{r create_table_merging_#OTUs_#reads_#samples_bacteria_non.norm, echo=FALSE}
tb16S_bacteria_OTUs_reads_samples_non.norm <- merge(class_summary_reads_per_class_16S_bacteria_non.norm_order, class_summary_otus_per_class_16S_bacteria_non.norm_order, by="row.names")

row.names(tb16S_bacteria_OTUs_reads_samples_non.norm)<-tb16S_bacteria_OTUs_reads_samples_non.norm[,1]
tb16S_bacteria_OTUs_reads_samples_non.norm<-tb16S_bacteria_OTUs_reads_samples_non.norm[,-1]
colnames(tb16S_bacteria_OTUs_reads_samples_non.norm)<-c("reads_per_class","OTUs_per_class")
tb16S_bacteria_OTUs_reads_samples_non.norm[1:5,1:2]

tb16S_bacteria_OTUs_reads_samples_non.norm<-tb16S_bacteria_OTUs_reads_samples_non.norm[order(tb16S_bacteria_OTUs_reads_samples_non.norm$reads_per_class, tb16S_bacteria_OTUs_reads_samples_non.norm$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
tb16S_bacteria_OTUs_reads_samples_non.norm["samples_per_class"] <- NA
tb16S_bacteria_OTUs_reads_samples_non.norm$samples_per_class<-c(9,9)

#table without "heterotropic_bacteria":
tb16S_bacteria_OTUs_reads_samples_non.norm <- tb16S_bacteria_OTUs_reads_samples_non.norm[-1,]
```

```{r OTUs_vs_reads_protists_relative_values_non.norm, echo=FALSE}

tb16S_protists_OTUs_reads_samples_non.norm_rel_abund <- tb16S_protists_OTUs_reads_samples_non.norm

tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$reads_per_class<-(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$reads_per_class*100)/colSums(tb16S_protists_OTUs_reads_samples_non.norm)[1]
tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$OTUs_per_class<-(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$OTUs_per_class*100)/colSums(tb16S_protists_OTUs_reads_samples_non.norm)[2]
tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$samples_per_class<-(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund$samples_per_class*100)/9

colSums(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund)
tb16S_protists_OTUs_reads_samples_non.norm_rel_abund
```

OTUs per class vs. samples in which they occur - PROTISTS [PLOT DESCRIPTION]

```{r stations_vs_OTUs_protists_16S_non.norm, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_OTUs_reads_samples_non.norm)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 120), breaks=c(25,50,75,100)) + labs(title="[15] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class - PROTISTS [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_16S_non.norm, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(tb16S_protists_OTUs_reads_samples_non.norm_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_OTUs_reads_samples_non.norm)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 60)) + labs(title="[16] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10")
```

OTUs per class vs. samples in which they occur - PROTISTS vs. CYANOBACTERIA [PLOT DESCRIPTION]

```{r merge_protists_and_bacteria_tb_non.norm, echo=FALSE}
tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm<-rbind(tb16S_protists_OTUs_reads_samples_non.norm,tb16S_bacteria_OTUs_reads_samples_non.norm)
```


```{r OTUs_vs_reads_protists_plus_bacteria_relative_values_non.norm, echo=FALSE}

tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund<-tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm

tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$reads_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$reads_per_class*100)/colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm)[1]

tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$OTUs_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$OTUs_per_class*100)/colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm)[2]

tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$samples_per_class<-(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund$samples_per_class*100)/9

colSums(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund)
tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund
```

```{r stations_vs_OTUs_protists_plus_bacteria_rel_abundance_non.norm, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 120)) + labs(title="[17] OTUs per class vs. samples in which they occur", x="samples (%)", y="OTUs (%)") + coord_trans(y = "log10")
```

Reads per class vs. OTUs per class - PROTISTS vs CYANOBACTERIA [PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_plus_bacteria_rel_abundance_non.norm, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples_non.norm_rel_abund)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 40)) + labs(title="[18] Reads per class vs. OTUs per class", x="OTUs (%)", y="reads (%)") + coord_trans(y = "log10") 
```

```{r cyanobacteria_histogram, echo=TRUE}
cyano_tb <- tb16_bacteria[which(tb16_bacteria$class_A != "heterotrophic_bacteria"),]

class_summary_reads_per_class_cyano<-aggregate(rowSums(cyano_tb[1:9]), list(cyano_tb$class_B), sum)
# count the different groups
class_summary_otus_per_class_cyano<-aggregate(rowSums(cyano_tb[1:9]), list(cyano_tb$class_B), length)

## READS PER CLASS ##
attach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano[order(-x),]
detach(class_summary_reads_per_class_cyano)
class_summary_reads_per_class_cyano_order

#fix column names
row.names(class_summary_reads_per_class_cyano_order)<-class_summary_reads_per_class_cyano_order[,1]
class_summary_reads_per_class_cyano_order<-class_summary_reads_per_class_cyano_order[c(-1)]
colnames(class_summary_reads_per_class_cyano_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano[order(-x),]
detach(class_summary_otus_per_class_cyano)
class_summary_otus_per_class_cyano_order

row.names(class_summary_otus_per_class_cyano_order)<-class_summary_otus_per_class_cyano_order[,1]
class_summary_otus_per_class_cyano_order<-class_summary_otus_per_class_cyano_order[c(-1)]
colnames(class_summary_otus_per_class_cyano_order)<-c("OTUs_per_class")
```

```{r cyanobacteria_histogram2, echo=TRUE}
# create_table_merging_#OTUs_#reads
cyano_OTUs_reads <- merge(class_summary_reads_per_class_cyano_order, class_summary_otus_per_class_cyano_order, by="row.names")
```

```{r cyanobacteria_histogram3, echo=TRUE}
row.names(cyano_OTUs_reads)<-cyano_OTUs_reads[,1]
cyano_OTUs_reads<-cyano_OTUs_reads[,-1]
colnames(cyano_OTUs_reads)<-c("reads_per_class","OTUs_per_class")
cyano_OTUs_reads[1:3,1:2]

cyano_OTUs_reads<-cyano_OTUs_reads[order(cyano_OTUs_reads$reads_per_class, cyano_OTUs_reads$OTUs_per_class, decreasing = T), c(1,2)]
cyano_OTUs_reads

#compute relative values
cyano_OTUs_reads_rel_abund <- cyano_OTUs_reads

cyano_OTUs_reads_rel_abund$reads_per_class<-(cyano_OTUs_reads_rel_abund$reads_per_class*100)/colSums(cyano_OTUs_reads)[1]
cyano_OTUs_reads_rel_abund$OTUs_per_class<-(cyano_OTUs_reads_rel_abund$OTUs_per_class*100)/colSums(cyano_OTUs_reads)[2]

colSums(cyano_OTUs_reads_rel_abund)
rownames(cyano_OTUs_reads_rel_abund) = c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")
cyano_OTUs_reads_rel_abund

cyano_OTUs_reads_rel_abund["cyano_group"]<-NA
cyano_OTUs_reads_rel_abund$cyano_group<-c("Prochlorococcus", "Synechococcus", "Other cyanobacteria")

cyano_OTUs_reads_rel_abund2<-read.table("/home/laura/Documents/TFM/home/data/MALASPINA/Malaspina_16S_Surface/amplicons_fraction/cyano_histograms_data.txt", head=TRUE)
cyano_OTUs_reads_rel_abund2

ggplot(cyano_OTUs_reads_rel_abund2, aes(x=group, y=value, fill=data)) + 
  geom_bar(position="dodge", stat="identity") +
  scale_y_continuous(limits=c(-1,100), breaks=c(25,50,75,100)) +
  scale_fill_manual(values=c("#000000", "#556670")) +
  labs(x="", y="% \n") + 
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16)) + 
  theme(legend.title = element_blank(), legend.position=c(0.85,0.85), legend.text=element_text(size=14)) +
  theme(axis.title.y = element_text(size = rel(1.5), angle = 0, vjust=0.97))
```
